skunkworks_crow/src/main/java/org/odk/share/bluetooth/BluetoothReceiver.java

```diff
@@ -33,24 +33,26 @@ public void onReceive(Context context, Intent intent) {
             return;
         }
 
-        BluetoothDevice bluetoothDevice = intent.getParcelableExtra(BluetoothDevice.EXTRA_DEVICE);
-        if (bluetoothDevice != null) {
-            Timber.d("BluetoothDevice: %s, Address: %s", bluetoothDevice.getName(), bluetoothDevice.getAddress());
-            switch (action) {
-                case BluetoothDevice.ACTION_FOUND:
+        switch (action) {
+            case BluetoothAdapter.ACTION_DISCOVERY_STARTED:
+                // if the discovery started, update the ui in activity.
+                Timber.d("bluetooth devices discovery started...");
```

Comment:
Hmm.. It seems `ACTION_DISCOVERY_STARTED` should write before `ACTION_FOUND`, I'm not sure, but it works now @lakshyagupta21 .